pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/OfflineClusterIntegrationTest.java

```diff
@@ -353,38 +354,52 @@ public void testTimeFunc()
-    assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(2).asText(), "abc");
-    assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(3).asText(), "today");
-    String nowColumnName = response.get("resultTable").get("dataSchema").get("columnNames").get(4).asText();
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(2).asText(), "oneHourAgoTs");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(3).asText(), "abc");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnNames").get(4).asText(), "today");
+    String nowColumnName = response.get("resultTable").get("dataSchema").get("columnNames").get(5).asText();
+    String oneHourAgoColumnName = response.get("resultTable").get("dataSchema").get("columnNames").get(6).asText();
     assertTrue(Long.parseLong(nowColumnName) > currentTsMin);
     assertTrue(Long.parseLong(nowColumnName) < currentTsMax);
+    assertTrue(Long.parseLong(oneHourAgoColumnName) > oneHourAgoTsMin);
+    assertTrue(Long.parseLong(oneHourAgoColumnName) < oneHourAgoTsMax);
 
     assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(0).asText(), "LONG");
     assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(1).asText(), "LONG");
-    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(2).asText(), "STRING");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(2).asText(), "LONG");
     assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(3).asText(), "STRING");
-    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(4).asText(), "LONG");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(4).asText(), "STRING");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(5).asText(), "LONG");
+    assertEquals(response.get("resultTable").get("dataSchema").get("columnDataTypes").get(6).asText(), "LONG");
 
     int first = response.get("resultTable").get("rows").get(0).get(0).asInt();
     long second = response.get("resultTable").get("rows").get(0).get(1).asLong();
-    String third = response.get("resultTable").get("rows").get(0).get(2).asText();
+    long third = response.get("resultTable").get("rows").get(0).get(2).asLong();
+    String fourth = response.get("resultTable").get("rows").get(0).get(3).asText();
     assertEquals(first, 1);
     assertTrue(second > currentTsMin);
     assertTrue(second < currentTsMax);
-    assertEquals(third, "abc");
-    String todayStr = response.get("resultTable").get("rows").get(0).get(3).asText();
+    assertTrue(third > oneHourAgoTsMin);
+    assertTrue(third < oneHourAgoTsMax);
```

Comment:
I think we should also add a test where ago() is used in comparison in the filter clause